sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorBuilder.java

```diff
@@ -0,0 +1,183 @@
+     * @return The updated {@link EventProcessorBuilder} instance.
+     */
+    public EventProcessorBuilder eventHubClient(EventHubAsyncClient eventHubAsyncClient) {
+        this.eventHubAsyncClient = eventHubAsyncClient;
+        return this;
+    }
+
+    /**
+     * Sets the consumer group name from which the {@link EventProcessor} should consume events.
+     *
+     * @param consumerGroup The consumer group name this {@link EventProcessor} should consume events.
+     * @return The updated {@link EventProcessorBuilder} instance.
+     */
+    public EventProcessorBuilder consumerGroup(String consumerGroup) {
+        this.consumerGroup = consumerGroup;
+        return this;
+    }
+
+    /**
+     * Sets the lambda function that will be used to process events from all partitions this {@link EventProcessor} is
+     * responsible for.
+     *
+     * @param processEvent The lambda function that will be used to process events.
+     * @return The updated {@link EventProcessorBuilder} instance.
+     */
+    public EventProcessorBuilder processEvent(Function<EventData, Mono<Void>> processEvent) {
+        this.processEvent = processEvent;
+        return this;
+    }
+
+    /**
+     * Sets the {@link PartitionManager} the {@link EventProcessor} will use for storing partition ownership and
+     * checkpoint information.
+     *
+     * <p>
+     * Users can, optionally, provide their own implementation of {@link PartitionManager} which will store ownership
+     * and checkpoint information. If not provided, the {@link EventProcessor} will first check for any available
+     * implementation of {@link PartitionManager} in the classpath and fallback to {@link InMemoryPartitionManager} if
+     * none found.
+     * </p>
+     *
+     * @param partitionManager Implementation of {@link PartitionManager}.
+     * @return The updated {@link EventProcessorBuilder} instance.
+     */
+    public EventProcessorBuilder partitionManager(PartitionManager partitionManager) {
+        this.partitionManager = partitionManager;
+        return this;
+    }
+
+    /**
+     * Sets the partition processor factory for creating new instance(s) of {@link PartitionProcessor}.
+     * <p>
+     * Use this to have finer control over processing events from partitions. This will also allow you to control the
+     * frequency of checkpointing.
+     * </p>
+     *
+     * @param partitionProcessorFactory The factory that creates new {@link PartitionProcessor} for each partition.
+     * @return The updated {@link EventProcessorBuilder} instance.
+     */
+    public EventProcessorBuilder partitionProcessorFactory(PartitionProcessorFactory partitionProcessorFactory) {
+        this.partitionProcessorFactory = partitionProcessorFactory;
+        return this;
+    }
+
+    /**
+     * Sets the initial event position. If this property is not set and if checkpoint for a partition doesn't exist,
```

Comment:
As you know, the first sentence is the most important sentence, as it gets shown in the summary table at the top of the JavaDoc. I suggest you consider something like this: 

```suggestion
     * Sets the initial event position, defaulting to {@link EventPosition#earliest()} if this property is not set.
```